src/SFA.DAS.Employer.PR.Web/Controllers/ChangePermissionsController.cs

```diff
@@ -0,0 +1,84 @@
+﻿using FluentValidation;
+using FluentValidation.AspNetCore;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Employer.PR.Domain.Interfaces;
+using SFA.DAS.Employer.PR.Domain.OuterApi.Permissions;
+using SFA.DAS.Employer.PR.Web.Authentication;
+using SFA.DAS.Employer.PR.Web.Constants;
+using SFA.DAS.Employer.PR.Web.Infrastructure;
+using SFA.DAS.Employer.PR.Web.Models;
+using SFA.DAS.Employer.PR.Web.Services;
+using SFA.DAS.Encoding;
+using System.Security.Claims;
+
+namespace SFA.DAS.Employer.PR.Web.Controllers;
+
+[Authorize(Policy = nameof(PolicyNames.HasEmployerOwnerAccount))]
+[Route("accounts/{employerAccountId}/providers/{ukprn}/changePermissions", Name = RouteNames.ChangePermissions)]
+public class ChangePermissionsController(IOuterApiClient _outerApiClient, IEncodingService _encodingService, IValidator<ChangePermissionsSubmitViewModel> _validator) : Controller
+{
+    [HttpGet]
+    public async Task<IActionResult> Index([FromRoute] string employerAccountId, [FromQuery] string legalEntityPublicHashedId,
+         [FromRoute] long ukprn, CancellationToken cancellationToken)
+    {
+
+        var legalEntityId = _encodingService.Decode(legalEntityPublicHashedId, EncodingType.PublicAccountLegalEntityId);
+
```

Comment:
Since we depend on the route parameters, we should validate if the combination of the two is valid for change permission, if it is not then we should take them to an error page. Currently there is option to take them PageNotFound, but  I am nor sure if that is the correct one.

Reply:
ok but this is not in the original sizing/spec so maybe needs to be a follow up task/story